media-libs/oidn/oidn-2.1.0.ebuild

```diff
@@ -0,0 +1,51 @@
+# Copyright 1999-2023 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+PYTHON_COMPAT=( python3_{9..11} )
+
+inherit cmake python-single-r1
+
+DESCRIPTION="Intel(R) Open Image Denoise library"
+HOMEPAGE="https://www.openimagedenoise.org/"
+
+if [[ ${PV} = *9999 ]]; then
+	EGIT_REPO_URI="https://github.com/OpenImageDenoise/oidn.git"
+	EGIT_BRANCH="master"
+	inherit git-r3
+else
+	SRC_URI="https://github.com/OpenImageDenoise/${PN}/releases/download/v${PV}/${P}.src.tar.gz -> ${P}.tar.gz"
+	KEYWORDS="~amd64 ~arm ~arm64 ~ppc64 ~x86"
+fi
+
+LICENSE="Apache-2.0"
+SLOT="0"
+REQUIRED_USE="
+	${PYTHON_REQUIRED_USE}
+	test? ( apps )
+"
+IUSE="apps cuda hip test"
+RESTRICT="!test? ( test )"
+
+RDEPEND="
+	${PYTHON_DEPS}
+	dev-cpp/tbb:=
+	dev-lang/ispc
+	cuda? ( dev-util/nvidia-cuda-toolkit )
+	hip? ( dev-util/hip )
+"
+DEPEND="${RDEPEND}"
+
+src_configure() {
+	local mycmakeargs=(
+		-DOIDN_DEVICE_CUDA=$(usex cuda)
+		-DOIDN_DEVICE_HIP=$(usex hip)
```

Comment:
Can't say for Cuda, but support of hip requires more flags. For me this works:

```
if use hip; then
	mycmakeargs+=(
		-DROCM_PATH="${EPREFIX}/usr"
		-DOIDN_DEVICE_HIP_COMPILER="$(get_llvm_prefix ${LLVM_MAX_SLOT})/bin/clang++"
	)
fi
```

However I'm testing not with hip-5.1.3 (which does not work for me at all!), I use hip 5.7.1 with patches from https://github.com/gentoo/gentoo/pull/33400.

Also after successful build there is a very ugly 
```
 * Detected file collision(s):
 * 
 *      /var/tmp/portage/media-libs/oidn-2.1.0/work/oidn-2.1.0_build/devices/hip/preinstall/lib64/libOpenImageDenoise_device_hip.so.2.1.0
 * 
 * Searching all installed packages for file collisions...
 * 
 * Press Ctrl-C to Stop
 * 
 * None of the installed packages claim the file(s).
 * 
 * Package 'media-libs/oidn-2.1.0' merged despite file collisions.
```
(it installs unnecessary copy of libOpenImageDenoise_device_hip.so.2.1.0 into `/var/tmp/portage/media-libs/oidn-2.1.0/work/oidn-2.1.0_build/devices/hip/preinstall/lib64/libOpenImageDenoise_device_hip.so.2.1.0`)

Reply:
My gut feeling is that we should probably patch their [CMake file](https://github.com/OpenImageDenoise/oidn/blob/master/devices/CMakeLists.txt#L87) to use the more flexible logic that is used in [Blender](https://projects.blender.org/blender/blender/src/branch/main/build_files/cmake/Modules/FindHIP.cmake).

We probably need to do the same thing with the duplicate install file.

Do you want me to look into this? I can't run HIP on my computer so I can't easily test that the resulting binaries/libraries actually work.

Reply:
Well, at least they provide CMake flag for ROCM path. But not for AMDGPU_TARGETS.

> Do you want me to look into this

Add this - https://github.com/DarkDefender/gentoo/pull/3. Again, I checked only hip 5.7.1 with patches, can't check CUDA.